src/ui/Facet/Facet.ts

```diff
@@ -1316,6 +1316,27 @@ export class Facet extends Component {
     this.queryStateModel.registerNewAttribute(this.lookupValueAttributeId, {});
 
     this.bind.onQueryState(MODEL_EVENTS.CHANGE, undefined, (args: IAttributesChangedEventArg) => this.handleQueryStateChanged(args));
+    this.listenToParentIfDependentFacet();
+  }
+
+  private listenToParentIfDependentFacet() {
```

Comment:
Ideally, we could have refactored all those methods and other parts of the components dealing with dependant facet inside a DependantFacetManager module, might have been much easier to test.

Reply:
Okay, I will move the relevant functions to a separate module. My thoughts are to implement it as follows:
```
private dependentFacetManager;

constructor() {
new DependentFacetManager(this);
}
```